src/core/or/connection_edge.c

```diff
@@ -883,14 +895,19 @@ export_hs_client_circuit_id(edge_connection_t *edge_conn,
   }
 
   /* Build the string */
-  tor_asprintf(&buf, "PROXY TCP6 %s:%x:%x %s %d %d\r\n",
+  tor_asprintf(&buf, "PROXY TCP6 %s:%x:%x:%x:%x:%x:%x %s %d %d\r\n",
                src_ipv6_prefix,
+               rp_addr >> 16, rp_addr & 0x0000ffff, rp_port,
```

Comment:
These should all be cast to uint16_t.

Reply:
Code removed in `5399de4`.

Reply:
All integers in this printf that are 4 hex digits wide should be cast to uint16_t. If we do the cast, then they can't possibly exceed 4 hex digits. If we don't, then any bugs produce bad format strings.

Reply:
Fixed in `8078900`.